### PR TITLE
Forgot Password Feature Added

### LINKED ISSUE #74: Forget Password Feature Not Added
### **Feature request**:
For Email and Password Auth, at Log-in, `Forget Password` Feature not added...I want to add it...

### **Is your feature request related to a problem? Please describe. (optional)**:
If user forget about password, they can reset it...

### **Describe the solution you'd like**:
When Click on `Forget Password`, a reset link will send to registed email....User Can Reset Mail from there

### **Describe alternatives you've considered**:
Mentioned Above

### **Describe how this feature will be useful to our readers**:
Mentioned Above

### **Additional context:**
As this navigation process, can't add anything

<hr/>

**Note:**
* If you want to work on an issue, you should check if it has already been assigned to anyone. 

I am also a GSSOC21 particiapant...Please assign me to here with add the required labels for that program

### PR DESCRIPTION
**Pull request by SamarpanCoder2002**
# Description:
Forgot Password Feature Added in Log-In

Fixes #74 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)

 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

# Screenshots / Video:
<p>

</p>

I am also a GSSOC21 participant.... Please raise the corresponding label of this program...

Hope you review it and merge it soon @ghackg11

### CODE REVIEW

**Comment by SamarpanCoder2002** [comment]
hello @ghackg11 .... Please review it and merge it ...Thanks

**Comment by sameeerkashyap** [comment]
@ghackg11 are we not using any separation of layers and state management like provider?  the above approach would be hard to maintain and scale further down the line

File: lib/components/logIn.dart [hunk 1]
```diff
@@ -35,232 +39,287 @@ class _LogInScreenState extends State<LogInScreen> {
...
+                                BoxShadow(
+                                  offset: Offset(-10.0, -200.0),
+                                  blurRadius: 0.0,
+                                  spreadRadius: 0.0,
+                                  color: Color(0x4f1e4d2b),
+                                )
+                              ],
+                              clipper: rclip(),
+                              child: Container(
+                                width: MediaQuery.of(context).size.width,
```
Comment (sameeerkashyap) [inline lib/components/logIn.dart]: Redundant code, try using MediaQuery.of(context) syntax only once. We might soon need to create a helper for this.

File: lib/components/logIn.dart [hunk 2]
```diff
@@ -35,232 +39,287 @@ class _LogInScreenState extends State<LogInScreen> {
...
-                            style: TextStyle(
-                              fontWeight: FontWeight.bold,
-                              color: Color(0xffF4F5F7),
-                              fontSize: 40,
+                          Align(
+                            alignment: Alignment.topLeft,
+                            child: Container(
+                              margin: EdgeInsets.only(
+                                left: MediaQuery.of(context).size.width * 0.1,
+                                top: MediaQuery.of(context).size.height * 0.1,
```
Comment (sameeerkashyap) [inline lib/components/logIn.dart]: same as above

File: lib/components/logIn.dart [hunk 3]
```diff
@@ -35,232 +39,287 @@ class _LogInScreenState extends State<LogInScreen> {
...
+                    Row(
+                      children: [
+                        Expanded(
+                          child: Padding(
+                            padding: EdgeInsets.only(left: 20.0, right: 20.0),
                             child: Container(
-                              //color: Colors.red,
+                              //color: Colors.yellow,
                               alignment: Alignment.center,
-                              width: MediaQuery.of(context).size.width * 0.4,
```
Comment (sameeerkashyap) [inline lib/components/logIn.dart:220]: same as above

**Review by sameeerkashyap — CHANGES_REQUESTED** [review]
Please refactor this code.

**Comment by ghackg11** [comment]
> @ghackg11 are we not using any separation of layers and state management like provider?  the above approach would be hard to maintain and scale further down the line 

Yup, this is an issue, you can create new issues regarding that, and manage it for better Scalability of the code.

**Comment by SamarpanCoder2002** [comment]
Hello @Sameerkash @ghackg11 I made changes in code according to all of you.... Please review it and merge it

File: lib/components/logIn.dart [hunk 4]
```diff
@@ -39,6 +39,7 @@ class _LogInScreenState extends State<LogInScreen> {
 
   @override
   Widget build(BuildContext context) {
+    final MediaQueryData mediaQueryData = MediaQuery.of(context);
```
Comment (sameeerkashyap) [inline lib/components/logIn.dart]: Why call size again multiple times? 

do this 
```
final MediaQueryData size = MediaQuery.of(context).size;
...
width: size.width,
 height: size.height,

```

**Review by sameeerkashyap — CHANGES_REQUESTED** [review]

**Comment by sameeerkashyap** [comment]
Also, I don't see any documentation in code

**Comment by SamarpanCoder2002** [comment]
> Also, I don't see any documentation in code

I made changes about 3 or 4 lines about forget password feature.... Variable and function name are user friendly to understand....Is it not enough??? @Sameerkash @ghackg11 .... Also, there is no other documentation before I start working on that....

**Comment by SamarpanCoder2002** [comment]
As per requirement of @Sameerkash ...... I added Documentation for Forgot Password Feature Which I add here...

Hope this will be fine @ghackg11 @Sameerkash

**Review by sameeerkashyap — APPROVED** [review]

**Comment by SamarpanCoder2002** [comment]
> As per requirement of @Sameerkash ...... I added Documentation for Forgot Password Feature Which I add here...
> 
> Hope this will be fine @ghackg11 @Sameerkash

Thanks for approve @Sameerkash ... Hope you review it and merge it soon @ghackg11 @vilsi12

**Comment by SamarpanCoder2002** [comment]
Thanks for merge @ghackg11 .... Nice to work in that project

**Comment by ghackg11** [comment]
> Thanks for merge @ghackg11 .... Nice to work in that project

Nice to work with you too:)

**Pull request merged**